modules/openapi-generator/src/main/resources/go/api.mustache

```diff
@@ -323,22 +312,18 @@ func (a *{{{classname}}}Service) {{nickname}}Execute(r {{#structPrefix}}{{&class
 {{/authMethods}}
 	req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
 	if err != nil {
-		executionError.error = err.Error()
-		return {{#returnType}}localVarReturnValue, {{/returnType}}nil, executionError
+		return {{#returnType}}localVarReturnValue, {{/returnType}}nil, err
 	}
 
 	localVarHTTPResponse, err := a.client.callAPI(req)
 	if err != nil || localVarHTTPResponse == nil {
-		executionError.error = err.Error()
-		return {{#returnType}}localVarReturnValue, {{/returnType}}localVarHTTPResponse, executionError
+		return {{#returnType}}localVarReturnValue, {{/returnType}}localVarHTTPResponse, err
 	}
 
 	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
 	localVarHTTPResponse.Body.Close()
-	localVarHTTPResponse.Body = _ioutil.NopCloser(bytes.NewBuffer(localVarBody))
```

Comment:
Good catch that the original PR also contains other fixes. 

My suggestion is to revert #8137 to start with and file new PRs afterwards for bug fixes and enhancements (like #8344 mentioned below).

What do you think?

Reply:
We can go that way, but this is a known issue. I think this can be done here to avoid a regression in the main branch.

Functionally, #8344 is equivalent to this revert minus this line. On the docs side it adds some clarification on how to get to the hidden methods on the return error. But I have no problem adding that in a later PR.

Reply:
Agreed it can be done here but I prefer to roll *all* changes in that PR to avoid any surprise and file separate PRs (bug fixes, enhancements, etc) for easier review.

Reply:
I've merged this PR and filed https://github.com/OpenAPITools/openapi-generator/pull/8439 to fix this issue.